src/functions/makeOrder.se

```diff
@@ -64,140 +60,129 @@ def makeOrder(sender: address, type, fxpAmount, fxpPrice, market, outcome, trade
+    log(type = logMakeOrder, market, sender, type, fxpPrice, fxpAmount, outcome, orderID, fxpMoneyEscrowed, fxpSharesEscrowed, block.timestamp, tradeGroupID)
     return(orderID)
 
-def placeAsk(sender: address, type, fxpAmount, fxpPrice, market, outcome):
+def placeAsk(sender: address, fxpAmount, fxpPrice, market, outcome):
     refund()
     if(msg.sender != self):
         ~invalid()
 
-    # Check if there are any "crossed orders" (i.e., bid orders at a price
-    # greater than or equal to the sell price).
-    bestBidID = ORDERS.getBestBid(market, outcome)
-    bestBidPrice = ORDERS.getPrice(bestBidID)
-    if(bestBidID and fxpPrice <= bestBidPrice and ORDERS.getAmount(bestBidID) >= safeFxpDiv(MIN_ORDER_VALUE, bestBidPrice)):
-        ~invalid()
-
-    moneyEscrowed = 0
-    sharesEscrowed = 0
+    fxpMoneyEscrowed = 0
+    fxpSharesEscrowed = 0
     event = MARKETS.getMarketEvent(market)
-    cumulativeScale = MARKETS.getCumulativeScale(market)
+    fxpCumulativeScale = MARKETS.getCumulativeScale(market)
 
     # Take shares in escrow if user has shares
-    sharesHeld = MARKETS.getParticipantSharesPurchased(market, sender, outcome)
-    if(sharesHeld):
-        minValue = 0
+    fxpSharesHeld = MARKETS.getParticipantSharesPurchased(market, sender, outcome)
+    if(fxpSharesHeld):
         if(EVENTS.getNumOutcomes(event) > 2):
-            minValue = ONE
+            fxpMinValue = ONE
         else:
-            minValue = EVENTS.getMinValue(event)
-        price = fxpPrice - minValue
-        cost = safeFxpMul(fxpAmount, price)
-        if(price > cumulativeScale or cost < MIN_ORDER_VALUE):
+            fxpMinValue = EVENTS.getMinValue(event)
+        fxpCostPerShare = fxpPrice - fxpMinValue
+        fxpCost = safeFxpMul(fxpAmount, fxpCostPerShare)
+        if(fxpCostPerShare > fxpCumulativeScale):
+            ~invalid()
+        if(fxpCost < MIN_ORDER_VALUE):
             ~invalid()
-        fill = min(sharesHeld, fxpAmount)
-        fxpAmount -= fill
-        sharesEscrowed += fill
-        MARKETS.modifyParticipantShares(market, sender, outcome, -fill, 0)
+        fxpFill = min(fxpSharesHeld, fxpAmount)
+        fxpAmount -= fxpFill
+        fxpSharesEscrowed += fxpFill
+        MARKETS.modifyParticipantShares(market, sender, outcome, -fxpFill, 0)
 
     # Take max minus price they want to sell at for the remainder of the amount
     # that wasn't able to be used up above
     if(fxpAmount):
-        maxValue = 0
         if(EVENTS.getNumOutcomes(event) > 2):
-            maxValue = ONE
+            fxpMaxValue = ONE
         else:
-            maxValue = EVENTS.getMaxValue(event)
-        costPerShareShorting = maxValue - fxpPrice
-        orderCost = safeFxpMul(fxpAmount, costPerShareShorting)
-        if(costPerShareShorting > cumulativeScale or orderCost < MIN_ORDER_VALUE):
+            fxpMaxValue = EVENTS.getMaxValue(event)
+        # FIXME what if fxpPrice > fxpMaxValue?
```

Comment:
It'd throw below since someone is trying to place an order that costs them negative money